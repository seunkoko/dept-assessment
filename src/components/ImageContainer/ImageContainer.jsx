// eslint-disable-next-line
import React from 'react';

import ViewCaseLink from '../common/ViewCaseLink';

import './ImageContainer.scss';

const ImageContainer = ({ data }) => {
	return (
    <>
      <div className="image-display">
        <img src={data.image} alt="img" />
        <div className="image-title">
          {data.title}
        </div>
        <div className="image-description">
          {data.description}
        </div>
        
        <ViewCaseLink
          id={data.id}
          moreDescription={data.moreDescription}
        />
      </div>
    </>
	);
};

export default ImageContainer;
