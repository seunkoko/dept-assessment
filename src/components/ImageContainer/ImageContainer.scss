@import "../../stylesheets/variables";

.image-display {
  letter-spacing: 1px;
}

.image-title {
  margin-top: 10px;
  color: $primary-gray;
  font-family: 'Teko-Regular';
  font-size: $small-font;
}

.image-description {
  font-family: 'Teko-Light';
  font-size: $medium-font;

  @media (max-width: $phone-width) {
    font-size: $semi-big-font;
  }
}
