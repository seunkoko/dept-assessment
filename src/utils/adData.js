import florensis from '../assets/images/florensis.png';

export const imageData = {
  id: 'adImage',
  image: florensis,
  title: 'Florensis',
  description: 'Rethinking the online ecosystem',
  moreDescription: 'Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.'
}

export const topFirstNews = {
  id: 'topFirstNews',
  title: 'MICROSOFT',
  news: 'Tapping into Ireland’s unique gaming culture and bringing a fresh flavour to their Xbox social media channels',
  moreDescription: ''
}

export const topSecondNews = {
  id: 'topSecondNews',
  title: 'O\'NEIL',
  news: 'Integrating existing content into O’Neill’s new e-commerce platform',
  moreDescription: ''
}

export const bottomFirstNews = {
  id: 'bottomFirstNews',
  title: 'BUTLINS',
  news: 'Enhancing customer experience for personalised holiday planning using real time data',
  moreDescription: ''
}

export const bottomSecondNews = {
  id: 'bottomSecondNews',
  title: 'VACANSESELECT',
  news: 'Predicting booking behavior for holidays with smart data, voice search and machine learning',
  moreDescription: ''
}
