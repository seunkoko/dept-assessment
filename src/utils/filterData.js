
export const workFilter = [
  {
    id: 1,
    name: 'Frontend Engineer',
  }, 
  {
    id: 2,
    name: 'Backend Engineer',
  }, 
  {
    id: 3,
    name: 'Data Scientist',
  },
];

export const industryFilter = [
  {
    id: 1,
    name: 'E-Commerce',
  }, 
  {
    id: 2,
    name: 'Medicine',
  }, 
  {
    id: 3,
    name: 'Robotics',
  },
];