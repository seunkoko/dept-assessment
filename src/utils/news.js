import bolcom from '../assets/images/bolcom.png';
import kempen from '../assets/images/kempen.png';
import philips from '../assets/images/philips.png';
import gemeentemuseum from '../assets/images/gemeentemuseum.png';
import lightning from '../assets/images/lightning.png';
import tui from '../assets/images/tui.png';
import chocomel from '../assets/images/chocomel.png';
import zalando from '../assets/images/zalando.png';
import jbl from '../assets/images/jbl.png';
import konin from '../assets/images/koninklijke-bibliotheek.png';
import liberty from '../assets/images/liberty-global.png';
import arla from '../assets/images/arla.png';

export const firstNewsData = [
  {
    id: 'bolcom',
    image: bolcom,
    title: 'BOLCOM',
    description: 'A summer island in the Netherlands',
    moreDescription: 'Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.'
  },
  {
    id: 'kempen',
    image: kempen,
    title: 'KEMPEN',
    description: 'Not some average banking website',
    moreDescription: 'Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.'
  },
  {
    id: 'phillips',
    image: philips,
    title: 'PHILIPS',
    description: 'Beautiful design meets innovative technology',
    moreDescription: 'Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.'
  },
  {
    id: 'gemeentemuseum',
    image: gemeentemuseum,
    title: 'GEMEENTEMUSEUM',
    description: 'A hundred years of Mondriaan & De Stijl',
    moreDescription: 'Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.'
  },
]

export const secondNewsData = [
  {
    id: 'lightning',
    image: lightning,
    title: 'BE LIGHTNING',
    description: 'Delivering clarity on a global scale',
    moreDescription: 'Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.'
  },
  {
    id: 'tui',
    image: tui,
    title: 'TUI',
    description: 'Swipe to find your next holiday destination',
    moreDescription: 'Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.'
  },
]

export const thirdNewsData = [
  {
    id: 'chocomel',
    image: chocomel,
    title: 'CHOCOMEL',
    description: 'A campaign for the limited edition letter packs',
    moreDescription: 'Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.'
  },
  {
    id: 'jbl',
    image: jbl,
    title: 'JBL',
    description: 'Live like a champion with Jerome Booteng',
    moreDescription: 'Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.'
  },
  {
    id: 'zalando',
    image: zalando,
    title: 'ZALANDO',
    description: 'Innovative SEO and content strategy for Zalando',
    moreDescription: 'Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.'
  },
  {
    id: 'konin',
    image: konin,
    title: 'KONINKLIJKE BIBLIOTHEEK',
    description: 'The search for the most influential book ever',
    moreDescription: 'Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.'
  },
]

export const fourthNewsData = [
  {
    id: 'liberty',
    image: liberty,
    title: 'LIBERTY GLOBAL',
    description: 'Delivering complex commerce solutions',
    moreDescription: 'Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.'
  },
  {
    id: 'arla',
    image: arla,
    title: 'ARLA',
    description: 'Swipe to find your next holiday destination',
    moreDescription: 'Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.'
  },
]
