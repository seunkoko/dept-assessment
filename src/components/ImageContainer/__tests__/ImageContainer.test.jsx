import React from 'react';

import { shallow } from 'enzyme';

import ImageContainer from '../ImageContainer';

import { imageData } from '../../../utils/adData';

describe('ImageContainer', () => {
	const component = shallow(<ImageContainer data={imageData} />);

	it('should render ImageContainer component', () => {
		expect(component).toMatchSnapshot();
  });
  
});