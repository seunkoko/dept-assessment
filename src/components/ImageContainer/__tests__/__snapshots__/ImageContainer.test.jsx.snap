// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageContainer should render ImageContainer component 1`] = `
<Fragment>
  <div
    className="image-display"
  >
    <img
      alt="img"
      src="test-file-stub"
    />
    <div
      className="image-title"
    >
      Florensis
    </div>
    <div
      className="image-description"
    >
      Rethinking the online ecosystem
    </div>
    <ViewCaseLink
      id="adImage"
      moreDescription="Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident."
    />
  </div>
</Fragment>
`;
